cpp/src/arrow/type.h

```diff
@@ -990,6 +990,71 @@ class ARROW_EXPORT LargeListType : public BaseListType {
   std::string ComputeFingerprint() const override;
 };
 
+/// \brief Type class for array of list views
+class ARROW_EXPORT ListViewType : public BaseListType {
+ public:
+  static constexpr Type::type type_id = Type::LIST_VIEW;
+  using offset_type = int32_t;
+
+  static constexpr const char* type_name() { return "list_view"; }
+
+  // ListView can contain any other logical value type
+  explicit ListViewType(const std::shared_ptr<DataType>& value_type)
```

Comment:
I'm being consistent with `ListType` and `LargeListType` (and all the parametric types). I would like to fix these constructors and type factories systematically instead of diverging from the existing pattern on this PR.

Reply:
I created an issue for this problem: https://github.com/apache/arrow/issues/37891